2_process.smk

```diff
@@ -0,0 +1,155 @@
+    return lake_sequence_files
+
+
+def save_sequences_summary(lake_sequence_files_input, summary_file):
+    """
+    Summarize the number of sequences with at least one temperature observation
+    for each lake, and save the result to csv
+
+    :param lake_sequence_files_input: the lake sequence files to summarize
+    :param summary_file: csv file with how many sequences are in each lake
+
+    """
+    sequence_counts = []
+    for sequences_file in lake_sequence_files_input:
+        num_sequences = np.load(sequences_file).shape[0] 
+        sequence_counts.append(num_sequences)
+    df_counts = pd.DataFrame(data={
+        'sequences_file': lake_sequence_files_input,
+        'num_sequences': sequence_counts
+    })
+    df_counts.to_csv(summary_file, index=False)
+
+
+def dynamic_filenames(wildcards):
+    """
+    Return the files that contain dynamic data that are needed to construct
+    sequences for a given lake.
+
+    This function also triggers four checkpoints:
+    1. fetch_mntoha_metadata to get lake_metadata.csv
+    2. unzip_archive for this lake's drivers
+    3. unzip_archive for this lake's clarity
+    3. unzip_archive for this lake's ice flags
+
+    :param wildcards: Snakemake wildcards (site_id from mntoha_lake_sequences).
+    :returns: List of 3 dynamic filenames: drivers, clarity, and ice flags
+
+    """
+    # make this function depend on fetch_mntoha_metadata
+    # needed because lake_metadata.csv is used to determine dynamic files
+    lake_metadata_file = checkpoints.fetch_mntoha_metadata.get(**wildcards).output[0]
+    lake_metadata = pd.read_csv(lake_metadata_file)
+    lake = lake_metadata.loc[lake_metadata['site_id']==wildcards.site_id].iloc[0]
+    # also make this function depend on unzip_archive
+    # needed to link unzipped files with unzip_archive rule
+    file_category = 'dynamic_mntoha'
+    drivers_directory = f'inputs_{lake.group_id}'
+    unzip_archive_drivers = checkpoints.unzip_archive.get(
+        file_category=file_category,
+        directory_name=drivers_directory
+    ).output[0]
+    clarity_directory = f'clarity_{lake.group_id}'
+    unzip_archive_clarity = checkpoints.unzip_archive.get(
+        file_category=file_category,
+        directory_name=clarity_directory
+    ).output[0]
+    ice_flags_directory = f'ice_flags_{lake.group_id}'
+    unzip_archive_ice_flags = checkpoints.unzip_archive.get(
+        file_category=file_category,
+        directory_name=ice_flags_directory
+    ).output[0]
+
+    # dynamic filenames
+    dynamic_dir = '1_fetch/out'
+    drivers_file = f'{dynamic_dir}/dynamic_mntoha/inputs_{lake.group_id}/{lake.meteo_filename}'
+    clarity_file = f'{dynamic_dir}/dynamic_mntoha/clarity_{lake.group_id}/gam_{lake.site_id}_clarity.csv'
+    ice_flags_file = f'{dynamic_dir}/dynamic_mntoha/ice_flags_{lake.group_id}/pb0_{lake.site_id}_ice_flags.csv'
```

Comment:
Since you've already defined parts of these paths as variables, I wonder if would make sense to edit these 3 lines to be:
```python
drivers_file = f'{dynamic_dir}/{file_category}/{drivers_directory}/{lake.meteo_filename}'
clarity_file = f'{dynamic_dir}/{file_category}/{clarity_directory}/gam_{lake.site_id}_clarity.csv'
ice_flags_file = f'{dynamic_dir}/{file_category}/{ice_flags_directory}/pb0_{lake.site_id}_ice_flags.csv'
```

Reply:
Yes, it does! In fact, the outputs from the calls to the `unzip_archive` checkpoint can be used, requiring fewer variables. Change made!